src/module_wrap.cc

```diff
@@ -207,6 +207,14 @@ void ModuleWrap::Evaluate(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+void ModuleWrap::Namespace(const FunctionCallbackInfo<Value>& args) {
+  auto iso = args.GetIsolate();
+  auto that = args.This();
+  ModuleWrap* obj = Unwrap<ModuleWrap>(that);
+  auto result = obj->module_.Get(iso)->GetModuleNamespace();
```

Comment:
`GetModuleNamespace()` terminates with a fatal error when `GetStatus()` isn't one of `kInstantiated`, `kEvaluating` or `kEvaluated`.

I believe I understand how the previous code enforced that invariant (by calling `.instantiate()` and `.evaluate()`, right?) but what ensures that now?

Reply:
nothing except when it is called, can add a check and throw though.

Reply:
made it throw after checking status